api/src/wfl/module/covid.clj

```diff
@@ -27,6 +28,37 @@
 (defn start-covid-workload!
   [])
 
+(defn ^:private snapshot-imported?
+  "Check if snapshot SNAPSHOT_ID in TDR has been imported to WORKSPACE
+  as SNAPSHOT_REFERENCE_ID and can be successfully fetched."
+  [{:keys [snapshot_id snapshot_reference_id] :as _source_details}
+   {:keys [workspace] :as _executor}]
+  (letfn [(fetch [] (rawls/get-snapshot-reference workspace
+                                                  snapshot_reference_id))]
+    (some? (and snapshot_id snapshot_reference_id (util/do-or-nil (fetch))))))
+
+(defn ^:private import-snapshot!
+  "Import snapshot SNAPSHOT_ID from TDR to WORKSPACE.
+  Use transaction TX to update DETAILS table with resulting reference id.
+  Throw if import fails."
+  [tx
+   {:keys [workload_id] :as _workload}
+   {:keys [details] :as _source}
+   {:keys [details_id snapshot_id] :as source_details}
+   {:keys [workspace] :as _executor}]
+  (letfn [(create! [] (rawls/create-snapshot-reference workspace snapshot_id))]
+    (if-let [reference-id (util/do-or-nil (:referenceId (create!)))]
```

Comment:
Why are you swallowing the exception here? If there's an error, we probably want to handle it further up the stack?

Reply:
I throw an exception if reference creation fails below, but you are right that I swallow the original exception:
 ```
(throw (ex-info "Rawls unable to create snapshot reference"
                      (assoc source_details :workspace workspace)))
```
Would you prefer no fussing with exceptions at this point in the code if we think it will be handled further up?  Or instead of swallowing, I could intercept and rethrow with additional information in the map on the executor, similar to how I approached this earlier on in this PR's life. 

At this point I lean towards the "no fuss" approach until things get stitched together.  Sorry about all the exception back-and-forth :(